src/tools/continuousintegration.rst

```diff
@@ -79,6 +79,101 @@ the button “Run pipeline” on a specific branch (i.e. master).
+-----------------------------------------------------------------------
+As part of the CI/CD process all teams are expected to collect and consolidate
+the necessary code health metrics under a file named :code:`ci-metrics.json` at
+the root of the repositories they have ownership over.
+
+The :code:`ci-metrics.json` file is expect to be created automatically as part
+of the CI pipeline by the teams by collecting the relevant information from the
+*unit tests*, *coverage*, *linting* and *build status*.
+**An important point to notice, is that :code:`ci-metrics.json` shouldn't exist
+as part of the repository, but, be created specifically as part of the CI
+pipeline.**
+The file must be created and properly populated before the start of the marked
+:code:`# START: Gitlab CI badges creation` in the **pages** stage of the CI
+pipeline (:code:`.gitlab-ci.yml` file).
+
+The metrics should be collected under the following structure:
+
+- build-status: *top level placeholder for the build process status*
+
+  - last: *placeholder about the last build process*
+
+    - status: (**string**) *failed or passed, stating the last build status*
+    - timestamp: (**float**) *the Unix timestamp with the date and time of the
+      last build status*
+
+  - green: *placeholder about the last green build process*
+
+    - timestamp: (**float**) *the Unix timestamp with the date and time of the
+      last successful build status*
+
+- coverage: *placeholder about the unit test coverage*
+
+  - percentage: (**float**) *the coverage percentage of the unit tests*
+
+- tests: *placeholder about the unit tests*
+
+  - passed: (**int**) *number of successful tests*
+  - failed: (**int**) *number of failed tests*
+  - total: (**int**) *total number of tests*
+
+- tests: *placeholder about the linting (static code analysis)*
+
+  - errors: (**int**) *number of linting errors*
+  - failures: (**int**) *number of linting failures* - this denotes a serious
+    error in the code that broke the linting process
+  - tests: (**int**) *total number of linting tests*
+
+:code:`ci-metrics.json` example:
+
+.. code-block:: json
+
+  {
+    "build-status": {
+        "last": {
+            "status": "failed",
+            "timestamp": 1568202193.0
+        },
+        "green": {
+            "timestamp": 1568112133.0
+        }
+    },
+    "coverage": {
+        "percentage": 60.00
+        },
+    "tests": {
+        "passed": 167,
+        "failed": 3,
+        "total": 170
+    },
+    "lint": {
+        "errors": 4,
+        "failures": 0,
+        "tests": 7
+    }
+}
+
+
+Besides the :code:`ci-metrics.json` the teams should provide the relevant
+output from the **test** and **linting** stage used to extract those metrics
+for the *unit test*, *coverage* and *linting* under specific files in JSON
+format for easy access as part of the published artifacts on the pages stage
+of the CI pipeline:
+
+- :code:`/public/reports/unit-tests.json`
```

Comment:
Can we have the absolute URLs as part of the JSON structure?

Reply:
Can you please clarify this question?

Reply:
this code is a relative url. It sould be great if this was an absolute URL that could be parsed from whatever tool in order to access the report as a URL